rotkehlchen/chain/evm/decoding/decoder.py

```diff
@@ -793,6 +794,29 @@ def _decode_transaction(
         if process_swaps:
             events = self._process_swaps(transaction=transaction, decoded_events=events)
 
+        if (
+                tx_receipt.tx_type == 4 and
+                transaction.from_address == transaction.to_address and
+                transaction.authorization_list is not None
+        ):  # if the event is an eip-7702 transaction, we need to add an informational event.
+            if (delegated_address := transaction.authorization_list[-1].delegated_address) == ZERO_ADDRESS:  # noqa: E501
```

Comment:
can it be empty by some kind of error? `-1` would raise index error here. Is it guaranteed somewhere else to not be empty?

Reply:
at every point, if the number of entries is zero, i set it to `None`. i can also add a `len(transaction.authorization_list) > 0` condition here if it makes sense